lib/Config/Hooks.php

```diff
@@ -132,8 +132,8 @@ function setupSubscriptionEvents() {
           'register_form',
           [$this->subscription_registration, 'extendForm']
         );
-        $this->wp->addAction(
-          'register_post',
+        $this->wp->addFilter(
+          'registration_errors',
```

Comment:
I understand the problem but is using `registration_errors` a good fix? Isn't this hook used rather for error processing or validation? Looking at https://usersinsights.com/wp/wp-content/uploads/2018/07/user-registration-hooks-2.png I have a question - could we use `user_register` or `register_new_user`? I.e. we would register after the user is really saved in WordPress DB?

Reply:
I don't think so. Maybe? I am not sure. 

We are already using `user_register` to catch any user creation and create the user in the `\MailPoet\Segments\WP::synchronizeUser` function. That hook can be created from anywhere where a user can be created

This is to only run the code in the registration form and get the data from it.

Reply:
Hm, I don't see the whole context, I just thought hooking on `registration_errors` with our `onRegister` action looks weird. If you think it's better to keep it in this hook, we can do it but it may look confusing to someone one day.

Reply:
I see what you mean. I don't think `user_register` is a better place and I don't know what would be better than `registration _errors`. I'll leave a comment.